Signal/src/AppDelegate.m

```diff
@@ -498,9 +495,9 @@ - (void)showImportUIForAttachment:(SignalAttachment *)attachment remainingRetrie
     viewController.attachment = attachment;
     UINavigationController *navigationController =
         [[UINavigationController alloc] initWithRootViewController:viewController];
-    [[[Environment getCurrent] homeViewController] presentTopLevelModalViewController:navigationController
-                                                                     animateDismissal:NO
-                                                                  animatePresentation:YES];
+    [SignalApp.sharedApp.homeViewController presentTopLevelModalViewController:navigationController
```

Comment:
Most of `Environment`'s functionality fell into three categories:

* Access to "main app" singletons.  
   * Some of these have been moved to `SignalApp`.
* Access to "main app and SAE" singletons.  
* UI Convenience functions (e.g. "open thread X").  These have been moved to SignalApp.

We could also convert some of the singletons to use the `sharedManager` pattern.